test/e2e/autotls/util.go

```diff
@@ -176,7 +96,8 @@ func disableNamespaceCertWithWhiteList(t *testing.T, clients *test.Clients, whit
 	}
 }
 
-func turnOnAutoTLS(t *testing.T, clients *test.Clients) context.CancelFunc {
+// TurnOnAutoTLS turns on Auto TLS feature.
+func TurnOnAutoTLS(t *testing.T, clients *test.Clients) context.CancelFunc {
```

Comment:
IMO our e2e tests (in Go) should not be manipulating our configmaps, this should be part of the test setup (could be Go).

I think a better separation of concerns here would be to have the `go test -tags=e2e` portion of the testing assume that auto-TLS has already been configured properly for the execution of the test.

What I want to test here is that the SUT can serve on TLS-terminated endpoints.  I think it is reasonable to expect that the SUT has been setup to do that however it deems appropriate.

To float a hypothetical question: What if we didn't need two different auto-TLS e2e tests?  What if we could run the same tests against two different configurations of cert-manager (HTTP01 and DNS01)?

Reply:
I followed the idea here and did the refactoring. Now there is only one E2E test case TestAutoTLS with 3 different configurations:
1. self-signed CA + per ksvc cert provision
2. self-signed CA + per namespace cert provision
3. http01 challenge

Please take a look. thanks!